backend/src/gpml/auth0_util.clj

```diff
@@ -5,11 +5,52 @@
 
 (def auth0-token (atom ""))
 
-(defn fetch-auth0-users [domain]
-  (client/get (format "%sapi/v2/users" domain)
-              {:content-type :json
-               :throw-exceptions false
-               :headers {:authorization (format "Bearer %s" @auth0-token)}}))
+(def ^:private ^:const default-filters-operator "AND")
+
+(defn- build-filters-query-string
```

Comment:
this functionality seems pretty interesting but not sure if we really need it for the current issue or perhaps we are overdesigning the solution 🤔 at the same time i don't see any unit test related so, my take on that, would be at least adding specific unit tests, then move it to another namespace. Even thinking if this logic could live in an external library... what do you think Lucas?

Reply:
Since we already had the `fetch-auth0-users` functionality I changed it a bit to accept filters. This way we can filter directly using the API when fetching users. However, to filter the users Auth0 only accepts a query parameter `q` which uses a Lucene-like syntax to specify the filters e.g. `q=email:lucas.sousa@magnet.coop`. So I decided to add a function to do that. But maybe you're right and this functionality is out of scope. Otherwise, we can move it to a different namespace as you say, and add tests to that specific function.

Reply:
@GeertSoet would you like to decide on the better option to follow up ?

Reply:
Yes. Do you think this could live in a `util.clj` namespace? Followed by its unit tests?

Reply:
I would say let's keep it easy for now, the goal is to get that ID in the JSON response. The other party knows how to query specific stakeholders already and didn't have an issue with it. 

But of course good to note in a "Magnet's view on the current DB/backend" that we can work on when we focus on refactoring :)

Reply:
Ok! Then I'll save that functionality in a snippet in my notes for the future and filter in the backend as we do in `list-auth0-verified-emails`.